near-sdk/src/types/vm_types.rs

```diff
@@ -1,9 +1,26 @@
-// TODO restrict to non-wasm32
-pub use near_vm_logic::types::{PromiseResult, ReturnData};
+#[cfg(not(target_arch = "wasm32"))]
+pub use near_vm_logic::types::ReturnData;
 
 //* Types from near_vm_logic
 
 pub type PublicKey = Vec<u8>;
 pub type PromiseIndex = u64;
 pub type ReceiptIndex = u64;
 pub type IteratorIndex = u64;
+
+// TODO(austinabell): This is not a great solution, but there is no need for the connection to
+// near_vm_logic and the serialize implementation is faulty. The type only needs to be kept as
+// is below because it is used only in a mocked blockchain interface.
+/// When there is a callback attached to one or more contract calls the execution results of these
+/// calls are available to the contract invoked through the callback.
+#[cfg(target_arch = "wasm32")]
+#[derive(Debug, PartialEq)]
+pub enum PromiseResult {
+    /// Current version of the protocol never returns `PromiseResult::NotReady`.
+    NotReady,
+    Successful(Vec<u8>),
+    Failed,
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+pub use near_vm_logic::types::PromiseResult;
```

Comment:
What if we remove `#[cfg(target_arch = "wasm32")]` and just always define our own `PromiseResult`, and use that in the `MockedBlockchain` interfaces, but, inside `MockedBlockchain`, convert `near_sdk::PromiseResult` into `near_vm_logic::types::PromiseResult`?

Reply:
I didn't because the only interface when it is used, the parameter is a `&[PromiseResult]` and I wanted to avoid creating a new vector just to pass it as a parameter. I'll change to this now though, because I do agree it could cause some jankyness, but I figured this would be abstracted away sometime soon.